### ISSUE TITLE
SpinBox: Increase increment/decrement speed on press and hold

### ISSUE THREAD

**Issue by FloVanGH** [slint-ui/slint#3052]
If the up or down button of the  `SpinBox` is press and hold the increment / decrement speed should be increased.

Mandatory to finish this issue it the completion of #3051

**Comment by FloVanGH** [comment]
> Hey! Also it would be nice to add another property (called e.g. `step`) which will allow adjusting the amount of a single increment/decrement click.

Thank you for your feedback. Makes totally sense from my point. I have extract your request in a separate issue #5281.

**Comment by Enyium** [comment]
At the end of a sequence of batched changes from holding the mouse button or arrow key, but also when `LineEdit` would call it (as long as there are real changes that weren't already communicated), the callback described in #5930 should be called. `SpinBox` would need to get this callback added to its API.

**Comment by Enyium** [comment]
I was about to create another issue for the following, since I thought this issue was only about being able to hold the button to perform multiple changes. But rereading the initial comment, it seems that this issue includes a proposal for acceleration. This is my issue text for it:

# `SpinBox` should accelerate

After #3052, `SpinBox` should provide a means to control acceleration of changing values when holding the mouse button or arrow key.

`step-size` could be changed to `step-sizes`, an array like `[1, 10, 100]`, `[1, 2, 5, 10]`, or even just `[1]`. Then, there could be an additional `acceleration-duration` property of type `duration` that allows to specify the time it takes to ramp up to the largest step size. I'm not sure whether a linear ramp-up curve would be best or some other curve.

**Comment by ogoffart** [comment]
Maybe now it can be implemented with the Timer
